ktor-client/ktor-client-plugins/ktor-client-auth/common/src/io/ktor/client/plugins/auth/providers/BearerAuthProvider.kt

```diff
@@ -24,17 +25,25 @@ public class BearerTokens(
     public val refreshToken: String
 )
 
+private typealias RefreshTokens =
+    suspend (client: HttpClient, response: HttpResponse, tokens: BearerTokens?) -> BearerTokens?
```

Comment:
Could you tell me why do we have the client in parameters and not in the receiver?

Reply:
3 parameters is quite long, should we try to group it inside some context-like class?

Reply:
1. `client.get("update-token").body<Token>()` looks more ideomatic than just `get("update-token").body<Token>()`
2. Yes, let's create a class.

Reply:
1. agree